### ISSUE TITLE
scroll lock not working when opening a new popup from popup

### ISSUE THREAD

**Issue by punthoofd07** [vast-engineering/jquery-popup-overlay#39]
Hi, 
When I open a new popup (when I click a button of an already opened popup), the scroll lock function is not working anymore, although the options are exact the same. 
Could you take a look at this please?
Thanks in advance for the great library! :-)

Kind regards,

**Comment by hlecorche** [comment]
Bug with scroll lock option and multiple popup instances : http://jsfiddle.net/3t74x6v0/3/

**Comment by pekanct** [comment]
i solved the bug, open the js of the script and:
1) 
add:
var fix_bar=0;
after:  var iOS = /(iPad|iPhone|iPod)/g.test(navigator.userAgent); // line 21

2)
add: fix_bar++;
after: if (options.scrolllock) { //line 337

3)
add: fix_bar--;
after: if (options.scrolllock) { //line 503

DONE :)

**Comment by dansurfrider** [comment]
Unfortunately I can't reconstruct pekanct's solution. Don't see how adding and substracting to a random variable can solve the issue anyway?!

**Comment by iorien** [comment]
Hi, I have the same problem.
In a page with a lot of content and vertical scroll, when opening the first popup with "scrolllock:true" the scroll disappears, when click a single button that closes the current popup and open a new one, the page scroll appears.

See example: http://jsfiddle.net/3t74x6v0/32/

I try the pekanct's solution unlucky.

**Comment by pvgoran** [comment]
You can try my fork https://github.com/pvgoran/jquery-popup-overlay that addresses problems with scrolllock in context of nested popups. (Not sure if it will work for your case, though.)

**Comment by vladimirsiljkovic** [comment]
Sorry for the late response.

pvgoran’s pull request with a fix for this issue was merged in https://github.com/vast-engineering/jquery-popup-overlay/pull/74 and available in v2.0.2

> Bug with scroll lock option and multiple popup instances : http://jsfiddle.net/3t74x6v0/3/

This should be working fine now. Z-index should work correctly as well (in v2.0.2).

pekanct's code was missing the last part, where the variable `lockedPopupsCount` would be used for a conditional statement. Pseudocode for that simple approach would be:

```
// init:
var lockedPopupsCount = 0;

// onOpen:
if (el.options.scrolllock) {
    lockedPopupsCount++;
}

// onclose:
if (el.options.scrolllock) {
	lockedPopupsCount--;
}
if (lockedPopupsCount === 0) {
    // re-enable scrolling by setting `overflow:visible` to html
}
```
